driver/src/test/java/org/neo4j/driver/internal/messaging/request/TransactionMetadataBuilderTest.java

```diff
@@ -119,11 +127,44 @@ void shouldIncludeNotificationConfig() {
                 null,
                 NotificationConfig.defaultConfig()
                         .enableMinimumSeverity(NotificationSeverity.WARNING)
-                        .disableCategories(Set.of(NotificationCategory.UNSUPPORTED)));
+                        .disableCategories(Set.of(NotificationCategory.UNSUPPORTED)),
+                Logging.none());
 
         var expectedMetadata = new HashMap<String, Value>();
         expectedMetadata.put("notifications_minimum_severity", value("WARNING"));
         expectedMetadata.put("notifications_disabled_categories", value(Set.of("UNSUPPORTED")));
         assertEquals(expectedMetadata, metadata);
     }
+
+    @ParameterizedTest
+    @ValueSource(longs = {1, 1_000_001, 100_500_000, 100_700_000, 1_000_000_001})
+    void shouldRoundUpFractionalTimeoutAndLog(long nanosValue) {
```

Comment:
Might also test if the round up doesn't happen and nothing is logged when the value is not fractional

Reply:
That bit is technically covered by the existing `shouldHaveCorrectMetadata` test. But I guess we can have a dedicated test as well.